### PR TITLE
fixed Config::get invalid keypath

### LINKED ISSUE #9: should throw custom error saying which field user wants to get when env var is not set

### PR DESCRIPTION
**Pull request by eric-bodhi**
Changed Config::get(const std::string& keyPath) to check if value is nullptr before casting Also changed ConfigDirectoryPathResolverTest::givenNotExistingPathInConfigDirectory_throws to make sure there's a nonexistent path

### CODE REVIEW

File: src/Config.cpp [hunk 1]
```diff
@@ -31,16 +31,24 @@ T Config::get(const std::string& keyPath)
...
     {
-        try
-        {
-            return std::to_string(std::any_cast<int>(value));
-        }
-        catch (const std::bad_any_cast& e)
+        // Check if the value is nullptr before casting
+        if (value.type() == typeid(std::nullptr_t))
         {
+            // Handle nullptr case by returning an empty string or another default value
```
Comment (cieslarmichal) [inline src/Config.cpp]: shouldnt it be outside of if statement that checks for std::string?

Comment (eric-bodhi) [reply src/Config.cpp]: are you talking about the comment or the actual code?

Comment (cieslarmichal) [reply src/Config.cpp]: Line 35

Comment (cieslarmichal) [reply src/Config.cpp]: Just asking, because it will only chceck for nullptr if std::string is requested

Comment (cieslarmichal) [reply src/Config.cpp]: Maybe it should check it always?

**Review by cieslarmichal — APPROVED** [review]

**Pull request merged**